hs/src/Reach/Eval/Core.hs

```diff
@@ -526,6 +527,61 @@ slToDL = \case
     yes = return . Just
     no = return Nothing
 
+getExportValArg :: DLExportValue -> Maybe DLArg
+getExportValArg (DLEV_Arg a) = Just a
+getExportValArg _ = Nothing
+
+slToDLExportVal :: SLVal -> App (Maybe DLExportValue)
+slToDLExportVal = \case
```

Comment:
The code you have duplicates a lot of the existing `slToDl` code and should not.

The function version would need to actually evaluate and collect the `DLStmts`

Reply:
This is still true

Reply:
I tried to refactor but I think the way the data is structured makes coalescing these functions much more complex to understand and work with . 

`DLExportVal` has the structure `DLArg | DLLargeArg` where `DLLargeArgs` contain `DLArgs`. Meanwhile, `DLArgExpr` has a flattened structure `DLArg | ...DLLargeArg`, where the constructors analogous to large args contain `DLArgExpr`. The result was just really messy and full of conditionals on whether it's producing an `ExportVal` or `DLArgExpr`

Reply:
Right now, there's

```
slToDL :: SLVal -> App (Maybe DLArgExpr)
```

I think there needs to be

```
newThing :: SLVal -> App (Maybe CoolNewThing)
cntToDL :: CoolNewThing -> App (Maybe DLArgExpr)
cntToEV :: CoolNewThing -> App (Maybe DLExportVal)
```

And maybe `CoolNewThing` just IS `DLArgExpr` and and `cntToDL` is really just `compileDLArgExpr`

Reply:
Bottom line: We can't have this duplication.

Reply:
If you want, I can experiment once you prototype